### PR TITLE
Implement DELETE for Policyfile types

### PR DESCRIPTION
**Pull request by danielsdeleo**
:construction: This only implements some of the required APIs :construction: 

Implements DELETE for policies, policy groups and policy revisions, along with some related GET APIs that will be used client-side for backup/undo functionality.

### CODE REVIEW

File: src/oc_erchef/apps/oc_chef_wm/src/oc_chef_wm_policy_group_policy_rev.erl [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+    stash_permissions_objects_authz_ids(Rest, UpdatedBaseState);
+stash_permissions_objects_authz_ids([{policy,AzID}|Rest], #base_state{resource_state = PolicyState} = State) ->
+    UpdatedPolicyState = PolicyState#policy_state{policy_authz_id = AzID},
+    UpdatedBaseState = State#base_state{resource_state = UpdatedPolicyState},
+    stash_permissions_objects_authz_ids(Rest, UpdatedBaseState).
+
+cleanup_orphaned_authz_ids(false, _RequestorId, _AuthzID) ->
+    ok;
+cleanup_orphaned_authz_ids(true, RequestorId, AuthzId) ->
+    oc_chef_authz:delete_resource(RequestorId, object, AuthzId),
```
Comment (marcparadise) [inline src/oc_erchef/apps/oc_chef_wm/src/oc_chef_wm_policy_group_policy_rev.erl]: Should we force the requestor as superuser here? If they ran into a permissions issue creating/updating things, it's possible that they won't be able to delete the IDs. 

Or are these IDs those that would have been generated in this request anyway?

File: src/oc_erchef/apps/oc_chef_wm/src/oc_chef_wm_policy_group_policy_rev.erl [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+    {{halt, 500}, Req, State#base_state{log_msg = ERROR}}.
+
+handle_update_result(ok, PolicyName, PolicyData, Req, State) ->
+    LogMsg = {updated, PolicyName},
+    ReqWithBody = chef_wm_util:set_json_body(Req, PolicyData),
+    {true, ReqWithBody, State#base_state{log_msg = LogMsg}};
+handle_update_result(ERROR, _PolicyName, _PolicyData, Req, State) ->
+    {{halt, 500}, Req, State#base_state{log_msg = ERROR}}.
+
+
```
Comment (marcparadise) [inline src/oc_erchef/apps/oc_chef_wm/src/oc_chef_wm_policy_group_policy_rev.erl]: extra blank lines here and below

**Comment by marcparadise** [comment]
I just had the couple of minor comments above.  I'm :+1: on this.

File: src/oc_erchef/apps/oc_chef_wm/src/oc_chef_wm_policy_group_policy_rev.erl (hunk 1 above)
Comment (danielsdeleo) [inline src/oc_erchef/apps/oc_chef_wm/src/oc_chef_wm_policy_group_policy_rev.erl]: Yup, these are generated in this request. This endpoint does create or update against two container/object types, so you might have permission for one and not the other. This gets called in `forbidden` (not using the one from `wm_base`) to clean up before returning the 403.

**Comment by stevendanna** [comment]
:+1:

**Pull request merged**